pandas/core/dtypes/missing.py

```diff
@@ -163,7 +163,10 @@ def _isna(obj, inf_as_na: bool = False):
     elif isinstance(obj, list):
         return _isna_ndarraylike(np.asarray(obj, dtype=object), inf_as_na=inf_as_na)
     elif hasattr(obj, "__array__"):
-        return _isna_ndarraylike(np.asarray(obj), inf_as_na=inf_as_na)
+        if hasattr(obj, "isna"):
+            return obj.isna()
```

Comment:
does this have any perf benefit?

Reply:
running asvs rn, will post

Reply:
What kind if object would hit this? Our own extension arrays are already handled above and passed to `_isna_ndarraylike`, which already calls `obj.isna()` in case of EAs.

So I *think* that means that only external objects we don't know that happen to have an `isna` method would end up here. So I don't think we should call `isna` in those cases, as there is no guarantee it returns the correct result.